engine-rest/engine-rest/src/main/java/org/camunda/bpm/engine/rest/exception/ExceptionLogger.java

```diff
@@ -20,36 +20,70 @@
 import java.io.PrintWriter;
 import java.io.StringWriter;
 import java.io.Writer;
-import java.util.logging.Level;
+import java.sql.SQLException;
 
+import org.camunda.bpm.engine.ProcessEnginePersistenceException;
 import org.camunda.commons.logging.BaseLogger;
 
+import static org.camunda.bpm.engine.impl.util.ExceptionUtil.PERSISTENCE_CONNECTION_ERROR_CLASS;
+
 public class ExceptionLogger extends BaseLogger {
 
   public static final String PROJECT_CODE = "ENGINE-REST";
   public static final String REST_API = "org.camunda.bpm.engine.rest.exception";
+
   public static final ExceptionLogger REST_LOGGER = BaseLogger.createLogger(ExceptionLogger.class,
-                                                                               PROJECT_CODE,
-                                                                               REST_API,
-                                                                               "HTTP");
+      PROJECT_CODE,
+      REST_API,
+      "HTTP"
+  );
 
   public void log(Throwable throwable) {
     Response.Status status = ExceptionHandlerHelper.getInstance().getStatus(throwable);
     int statusCode = status.getStatusCode();
 
-    if ( statusCode >= 500) {
-      logWarn(String.valueOf(statusCode), getStackTrace(throwable));
-    } else {
+    if (statusCode < 500) {
       logDebug(String.valueOf(statusCode), getStackTrace(throwable));
+      return;
     }
+
+    logInternalServerAndOtherStatusCodes(throwable, statusCode);
   }
 
-  protected String getStackTrace(Throwable aThrowable) {
+  protected void logInternalServerAndOtherStatusCodes(Throwable throwable, int statusCode) {
+
+    if (isPersistenceConnectionError(throwable)) {
+      logError(String.valueOf(statusCode), getStackTrace(throwable));
+      return;
+    }
 
-    final Writer      result      = new StringWriter();
+    logWarn(String.valueOf(statusCode), getStackTrace(throwable));
+  }
+
+  protected String getStackTrace(Throwable throwable) {
+    final Writer result = new StringWriter();
     final PrintWriter printWriter = new PrintWriter(result);
-    aThrowable.printStackTrace(printWriter);
+    throwable.printStackTrace(printWriter);
 
     return result.toString();
   }
-}
\ No newline at end of file
+
+  protected boolean isPersistenceConnectionError(Throwable throwable) {
+    boolean isPersistenceException = (throwable instanceof ProcessEnginePersistenceException);
+
+    if (!isPersistenceException) {
+      return false;
+    }
+
+    SQLException sqlException = (SQLException) throwable.getCause().getCause();
+
+    if (sqlException == null) {
+      return false;
+    }
```

Comment:
🔧 I would suggest to include a type check here. If the cause is not a `SQLException`, we'll unnecessarily run into unexpected `ClassCastException`s here.

Reply:
The design, discussed also with tassilo is meant to always include a SQL exception as a second cause whenever there is a `ProcessEnginePersistenceException`. However, i also feel some insecurity with the lack of type check. Will add it just to make sure.

Reply:
For now, i believe silently ignoring it and continue is ok. I would recommend to log it in debug level but the whole logic is applied to log in the end the persistence exception. So logging it in debug would actually beat the initial purpose.

Reply:
Yeah, I agree 👍 It's rather unlikely that this happens in the first place anyway.